src/main/java/gov/cms/mat/cql_elm_translation/service/CqlConversionService.java

```diff
@@ -180,4 +198,138 @@ private String formatMessage(CqlCompilerException e) {
         e.getSeverity() != null ? e.getSeverity().name() : null,
         e.getMessage());
   }
+
+  public void processNoContextError(CqlTranslator cqlTranslator, String cql) {
+    if (StringUtils.isNotBlank(cql)
+        && cqlTranslator.getTranslatedLibrary().getLibrary().getContexts() == null) {
+      VersionedIdentifier identifier =
+          cqlTranslator.getTranslatedLibrary().getLibrary().getIdentifier();
+
+      List<Integer> allLines = new ArrayList<>();
+      int startLineForContext = 0;
+      Library.Statements statements =
+          cqlTranslator.getTranslatedLibrary().getLibrary().getStatements();
+      if (statements != null) {
+        List<ExpressionDef> defs = statements.getDef();
+        allLines = getStartLinesFromStatements(defs);
+        startLineForContext = allLines.size() > 0 ? allLines.get(0) - 1 : 0;
+      } else {
+        allLines.addAll(getUsingEndLines(cqlTranslator));
+        allLines.addAll(getParameterEndLines(cqlTranslator));
+        allLines.addAll(getValueSetsEndLines(cqlTranslator));
+        allLines.addAll(getCodeSystemEndLines(cqlTranslator));
+        allLines.addAll(getCodeEndLines(cqlTranslator));
+        allLines.sort(Comparator.reverseOrder());
+        startLineForContext = allLines.size() > 0 ? allLines.get(0) + 1 : 0;
+      }
+      log.debug("Missing context at line: " + startLineForContext);
+
+      cqlTranslator
+          .getExceptions()
+          .add(new MissingContextException(identifier, startLineForContext));
+    }
+  }
+
+  protected List<Integer> getStartLinesFromStatements(List<ExpressionDef> defs) {
+    List<Integer> startLines = new ArrayList<>();
+    if (CollectionUtils.isNotEmpty(defs)) {
+      for (ExpressionDef def : defs) {
```

Comment:
Are we looping over all the definitions because the definitions are in random order? trying to understand why we need to iterate over all definitions.

Reply:
@adongare It seems to me that "context Patient" needs to be above definitions otherwise there will be error?